include/gsl/gsl_assert

```diff
@@ -110,10 +125,7 @@ namespace details
 #if defined(GSL_TERMINATE_ON_CONTRACT_VIOLATION)
 
     template <typename Exception>
-#if defined(__clang__) || defined(__GNUC__)
-#else
-    [[gsl::suppress(f.6)]]
-#endif
+    GSL_SUPPRESS(f.6) // NO-FORMAT: attribute
```

Comment:
Couldn't we use a conditional noexcept for these cases, based on the throw/terminate build-mode? Yes, it might involve a macro, but probably less directly and would be no worse than suppressions. But I may be missing something else here.